src/modules/reservation.js

```diff
@@ -0,0 +1,126 @@
+import { displayReservations } from './displayReservations.js';
+// eslint-disable-next-line no-inner-declarations
+export async function popupReservation(show) {
+  if (show && show.show) {
+    const overlay = document.createElement('div');
+    overlay.className = 'overlay';
+
+    const popup = document.createElement('div');
+    popup.className = 'popup';
+    const closeButtonContainer = document.createElement('div');
+    closeButtonContainer.className = 'close-button-container';
+
+    const closeButton = document.createElement('button');
+    closeButton.textContent = 'Close';
+    closeButton.className = 'close-button';
+    closeButton.onclick = () => {
+      document.body.removeChild(overlay);
+    };
+    const content = document.createElement('div');
+    const img = document.createElement('img');
+    img.src = show.show.image ? show.show.image.medium : 'placeholder.jpg';
+    img.alt = show.show.name;
+
+    const h2 = document.createElement('h2');
+    h2.textContent = show.show.name;
+
+    const summary = document.createElement('p');
+    summary.innerHTML = show.show.summary;
+
+    const reservationForm = document.createElement('form');
+    reservationForm.className = 'reservation-form';
+
+    const nameInput = document.createElement('input');
+    nameInput.type = 'text';
+    nameInput.placeholder = 'Name';
+
+    const startDateInput = document.createElement('input');
+    startDateInput.type = 'date';
+    startDateInput.placeholder = 'Start Date';
+
+    const endDateInput = document.createElement('input');
+    endDateInput.type = 'date';
+    endDateInput.placeholder = 'End Date';
+
+    const reserveButton = document.createElement('button');
+    reserveButton.textContent = 'Reserve';
+    const reservationsContainer = document.createElement('div');
+    reservationsContainer.className = 'reservations-container';
+
+    const totalReservations = document.createElement('p');
+    totalReservations.className = 'total-reservations';
+    // eslint-disable-next-line no-inner-declarations
```

Comment:
Let's not disable linters, whenever linters highlight errors, it's our job to correct those errors.